src/dattobd.c

```diff
@@ -77,22 +88,89 @@ static loff_t noop_llseek(struct file *file, loff_t offset, int origin){
 }
 #endif
 
+
+#ifndef HAVE_STRUCT_PATH
+struct path {
+	struct vfsmount *mnt;
+	struct dentry *dentry;
+};
+#define dattobd_get_dentry(f) f->f_dentry
+#define dattobd_get_mnt(f) f->f_vfsmnt
+#define dattobd_d_path(path,page_buf,page_size) d_path(path->dentry,path->mnt,page_buf,page_size)
+#define dattobd_get_nd_dentry(nd) nd.dentry
+#else
+#define dattobd_get_dentry(f) f->f_path.dentry
+#define dattobd_get_mnt(f) f->f_path.mnt
+#define dattobd_d_path(path,page_buf,page_size) d_path(path,page_buf,page_size)
+#define dattobd_get_nd_dentry(nd) nd.path.dentry
+#endif
+
+#ifndef HAVE_BLKDEV_GET_BY_PATH
+struct block_device *dattobd_lookup_bdev(const char *pathname, fmode_t mode) {
+	int r;
+	struct block_device *retbd;
+	struct nameidata nd;
+	struct inode *inode;
+	dev_t dev;
+
+	if ((r = path_lookup(pathname, LOOKUP_FOLLOW, &nd)))
+		goto fail;
+
+	inode = dattobd_get_nd_dentry(nd)->d_inode;
+	if (!inode) {
+		r = -ENOENT;
+		goto fail;
+	}
+
+	if (!S_ISBLK(inode->i_mode)) {
+		r = -ENOTBLK;
+		goto fail;
+	}
+	LOG_DEBUG("dattobd_lookup_bdev inode from path_lookup: %p", inode);
```

Comment:
get rid of the `LOG_DEBUG()`s in the version abstraction section so that we don't get different debug messages for different kernels